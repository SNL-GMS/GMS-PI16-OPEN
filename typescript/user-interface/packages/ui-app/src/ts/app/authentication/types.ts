import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
