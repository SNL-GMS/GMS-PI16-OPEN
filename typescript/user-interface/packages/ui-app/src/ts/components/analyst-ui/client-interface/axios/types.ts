/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
