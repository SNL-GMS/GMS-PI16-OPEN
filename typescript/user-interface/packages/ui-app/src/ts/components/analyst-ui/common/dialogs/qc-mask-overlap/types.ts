import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
