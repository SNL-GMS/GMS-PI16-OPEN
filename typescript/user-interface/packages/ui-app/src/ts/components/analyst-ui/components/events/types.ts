import { GQLTypes } from '@gms/common-graphql';
import { CommonTypes, EventTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { ChildProps, MutationFunction } from 'react-apollo';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction;
  saveEvent: MutationFunction;
  saveAllModifiedEvents: MutationFunction;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analysisMode: WorkflowTypes.AnalysisMode;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  GQLTypes.EventsInTimeRangeQueryProps &
  GQLTypes.DefaultStationsQueryProps &
  GQLTypes.SignalDetectionsByStationQueryProps &
  GQLTypes.WorkspaceStateProps;
