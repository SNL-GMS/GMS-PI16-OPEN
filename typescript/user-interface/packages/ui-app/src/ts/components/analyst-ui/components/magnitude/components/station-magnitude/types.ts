import { EventTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { MutationFunction } from 'react-apollo';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { AmplitudesByStation } from '../../types';

/**
 * StationMagnitude State
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface StationMagnitudeState {
  computeNetworkMagnitudeSolutionStatus: Immutable.Map<
    string,
    [{ stationName: string; rational: string }]
  >;
}

export interface MagnitudeAndSdData {
  magSolution: EventTypes.NetworkMagnitudeSolution;
  sdData: StationMagnitudeSdData;
}

/**
 * options that can be passed to ag grid
 */
export interface Options {
  alignedGrids: any[];
}

/**
 * StationMagnitude Props
 */
export interface StationMagnitudeProps {
  options?: Options;
  amplitudesByStation: AmplitudesByStation[];
  historicalMode: boolean;
  selectedSdIds: string[];
  locationSolution: EventTypes.LocationSolution;
  displayedMagnitudeTypes: AnalystWorkspaceTypes.DisplayedMagnitudeTypes;
  computeNetworkMagnitudeSolution: MutationFunction;
  openEventId: string;

  setSelectedSdIds(ids: string[]): void;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<[{ stationName: string; rational: string }]>;
}

export interface StationMagnitudeSdData {
  channel: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  signalDetectionId: string;
  time: number;
  stationName: string;
  flagForReview: boolean;
}

export interface MagnitudeDataForRow {
  channel: string;
  signalDetectionId: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  flagForReview: boolean;
  defining: boolean;
  mag: number;
  res: number;
  hasMagnitudeCalculationError: boolean;
  computeNetworkMagnitudeSolutionStatus: string;
}

/**
 * Table row object for station magnitude
 */
export interface StationMagnitudeRow {
  id: string;
  dataForMagnitude: Map<EventTypes.MagnitudeType, MagnitudeDataForRow>;
  station: string;
  dist: number;
  azimuth: number;
  selectedSdIds: string[];
  historicalMode: boolean;
  azimuthTooltip: string;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<void>;
}
export interface StationMagAndSignalDetection {
  stationName: string;
  magnitudeAndSdData: Map<EventTypes.MagnitudeType, MagnitudeAndSdData>;
}
