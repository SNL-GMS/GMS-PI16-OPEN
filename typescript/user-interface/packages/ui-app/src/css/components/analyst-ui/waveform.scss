.station-name {
  white-space: nowrap;

  &__channel-name {
    color: $gms-prominent;
    font-size: 0.8em;
  }
}
.radio-dropdown {
  padding: 0.5rem 1rem;
  label {
    margin: 0 -0.5em;
    height: 2.125em;
    display: flex;
    align-items: center;
    &:hover {
      filter: brightness(0.8);
    }
    * + .bp3-html-select {
      margin-left: 1ch;
    }
  }
}
