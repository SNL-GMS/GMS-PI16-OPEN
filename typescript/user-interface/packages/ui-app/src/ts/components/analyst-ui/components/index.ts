export { ReduxApolloAzimuthSlowness as AzimuthSlowness } from './azimuth-slowness/azimuth-slowness-container';
export { Events } from './events';
export { History } from './history';
export { Location } from './location';
export { Magnitude } from './magnitude';
export { IANMap } from './map';
export { ReduxApolloMap as Map } from './map/map-container';
export { SignalDetections } from './signal-detections';
export { StationPropertiesComponent, StationPropertiesPanel } from './station-properties';
export { ReactQueryWaveformDisplay as WaveformDisplay } from './waveform';
export { Workflow } from './workflow';
