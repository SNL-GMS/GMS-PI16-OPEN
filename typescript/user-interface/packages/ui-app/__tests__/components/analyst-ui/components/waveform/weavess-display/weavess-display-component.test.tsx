/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import { CommonTypes, WorkflowTypes } from '@gms/common-model';
import { readJsonData, toEpochSeconds } from '@gms/common-util';
import { AnalystWorkspaceOperations, AnalystWorkspaceTypes, createStore } from '@gms/ui-state';
import { WeavessTypes } from '@gms/weavess-core';
import { mount } from 'enzyme';
import * as path from 'path';
import * as React from 'react';
import { Provider } from 'react-redux';

import { AmplitudeScalingOptions } from '../../../../../../src/ts/components/analyst-ui/components/waveform/components/waveform-controls/scaling-options';
import { WeavessContext } from '../../../../../../src/ts/components/analyst-ui/components/waveform/weavess-context';
import { WeavessDisplay } from '../../../../../../src/ts/components/analyst-ui/components/waveform/weavess-display';
import { WeavessDisplayProps } from '../../../../../../src/ts/components/analyst-ui/components/waveform/weavess-display/types';
import { QcMaskType } from '../../../../../../src/ts/components/analyst-ui/config/system-config';
import { BaseDisplayContext } from '../../../../../../src/ts/components/common-ui/components/base-display';

const basePath = path.resolve(__dirname, '../__data__');
const waveformState: any = readJsonData(path.resolve(basePath, 'waveformState.json'));

jest.mock(
  '../../../../../../src/ts/components/analyst-ui/client-interface/axios/queries/waveform-query',
  () => {
    const actualWaveformQuery = jest.requireActual(
      '../../../../../../src/ts/components/analyst-ui/client-interface/axios/queries/waveform-query'
    );
    return {
      ...actualWaveformQuery,
      getBoundaries: jest.fn(() => ({
        topMax: 100,
        bottomMax: -100,
        channelAvg: 0,
        offset: 100,
        channelSegmentId: 'TEST',
        samplesCount: 100
      }))
    };
  }
);

const initialConfiguration: WeavessTypes.Configuration = {
  shouldRenderWaveforms: true,
  shouldRenderSpectrograms: false,
  hotKeys: {
    amplitudeScale: 'KeyS',
    amplitudeScaleSingleReset: 'Alt+KeyS',
    amplitudeScaleReset: 'Alt+Shift+KeyS',
    maskCreate: 'KeyM'
  },
  defaultChannel: {
    disableMeasureWindow: true,
    disableSignalDetectionModification: false,
    disableMaskModification: true
  },
  nonDefaultChannel: {
    disableMeasureWindow: true,
    disableSignalDetectionModification: true,
    disableMaskModification: false
  }
};
const startTimeSecs = toEpochSeconds('2010-05-20T22:00:00.000Z');
const endTimeSecs = toEpochSeconds('2010-05-20T23:59:59.000Z');

const qcMaskVersion = {
  category: 'REJECTED',
  channelSegmentIds: [],
  startTime: 0,
  endTime: 1,
  rationale: 'just because',
  type: QcMaskType.SPIKE,
  version: 'version'
};
describe('weavess display', () => {
  const weavessProps: WeavessDisplayProps = {
    weavessProps: {
      startTimeSecs: waveformState.viewableInterval.startTimeSecs,
      endTimeSecs: waveformState.viewableInterval.endTimeSecs,
      showMeasureWindow: false,
      initialZoomWindow: {
        startTimeSecs,
        endTimeSecs
      },
      defaultZoomWindow: {
        startTimeSecs: waveformState.viewableInterval.startTimeSecs,
        endTimeSecs: waveformState.viewableInterval.endTimeSecs
      },
      stations: [],
      events: undefined,
      measureWindowSelection: {
        channel: undefined,
        endTimeSecs: undefined,
        isDefaultChannel: undefined,
        startTimeSecs: undefined,
        stationId: undefined
      } as WeavessTypes.MeasureWindowSelection,
      selections: { signalDetections: [], channels: [] },
      initialConfiguration,
      customMeasureWindowLabel: undefined,
      flex: false
    },
    defaultWaveformFilters: [],
    defaultStations: [],
    eventsInTimeRange: [],
    signalDetectionsByStation: [],
    qcMasksByChannelName: [
      {
        currentVersion: qcMaskVersion,
        channelName: 'chanName',
        id: 'masky',
        qcMaskVersions: [qcMaskVersion]
      }
    ],
    measurementMode: {
      mode: AnalystWorkspaceTypes.WaveformDisplayMode.DEFAULT,
      entries: undefined
    } as AnalystWorkspaceTypes.MeasurementMode,
    defaultSignalDetectionPhase: CommonTypes.PhaseType.P,
    setMeasurementModeEntries: AnalystWorkspaceOperations.setMeasurementModeEntries,
    analysisMode: undefined,
    createEvent: undefined,
    createQcMask: undefined,
    createSignalDetection: undefined,
    currentOpenEventId: undefined,
    currentTimeInterval: undefined,
    rejectQcMask: undefined,
    rejectSignalDetection: undefined,
    sdIdsToShowFk: undefined,
    selectedSdIds: undefined,
    selectedStationIds: undefined,
    setEventSignalDetectionAssociation: undefined,
    updateQcMask: undefined,
    updateSignalDetection: undefined,
    glContainer: undefined,
    setMode: undefined,
    setOpenEventId: undefined,
    setSdIdsToShowFk: undefined,
    setSelectedSdIds: undefined,
    setSelectedStationIds: undefined,
    amplitudeScaleOption: AmplitudeScalingOptions.FIXED,
    fixedScaleVal: 26
  };

  const timeRange: CommonTypes.TimeRange = {
    startTimeSecs,
    endTimeSecs
  };
  const workflowState = {
    timeRange,
    analysisMode: WorkflowTypes.AnalysisMode.SCAN,
    stationGroup: null,
    openIntervalName: null,
    openActivityNames: []
  };

  const store = createStore();
  store.getState().analystWorkspaceState.workflowState = workflowState;
  let weavessRef: any = {
    waveformPanelRef: {
      stationComponentRefs: {
        values: () => [
          {
            props: { station: { name: 'AAK.BHZ' }, nonDefaultChannels: ['AAK.AAK.BHZ'] },
            state: {}
          },
          {
            props: { station: { name: 'AFI.BHZ', nonDefaultChannels: ['AFI.AFI.BHZ'] } },
            state: { expanded: true }
          }
        ]
      }
    },
    getCurrentViewRangeInSeconds: jest.fn(() => ({ startTimeSecs: 0, endTimeSecs: 1000 }))
  };
  const waveform = (
    <Provider store={store}>
      <BaseDisplayContext.Provider
        value={{
          glContainer: undefined,
          widthPx: 1920,
          heightPx: 1080
        }}
      >
        <WeavessContext.Provider
          value={{
            weavessRef,
            setWeavessRef: ref => {
              weavessRef = ref;
            }
          }}
        >
          <WeavessDisplay {...weavessProps} />
        </WeavessContext.Provider>
      </BaseDisplayContext.Provider>
    </Provider>
  );

  const wrapper = mount(waveform);
  const weavessDisplayComponent: any = wrapper.find('WeavessDisplayComponent').instance();

  test('can mount waveform panel and interact with it', async () => {
    expect(wrapper).toMatchSnapshot();

    // Lets try and exercise somethings
    weavessDisplayComponent.props.defaultStations[0] = {
      name: 'AAK.BHZ',
      allRawChannels: [{ name: 'AAK.BHZ' }, { name: 'AAK.AAK.BHZ' }]
    };
    weavessDisplayComponent.props.defaultStations[1] = {
      name: 'AFI.BHZ',
      allRawChannels: [{ name: 'AFI.BHZ' }, { name: 'AFI.AFI.BHZ' }]
    };
    weavessDisplayComponent.onUpdateChannelMarker();
    const event: Partial<React.MouseEvent<HTMLDivElement>> = {
      preventDefault: jest.fn(),
      shiftKey: true
    };
    weavessDisplayComponent.onMaskClick(
      event,
      'AAK.AAK.BHZ',
      [weavessProps.qcMasksByChannelName[0].id],
      false
    );
    event.shiftKey = false;
    weavessDisplayComponent.onMaskClick(
      event,
      'AAK.AAK.BHZ',
      [weavessProps.qcMasksByChannelName[0].id],
      true
    );
    weavessProps.qcMasksByChannelName.push(weavessProps.qcMasksByChannelName[0]);
    weavessDisplayComponent.onMaskClick(
      event,
      'AAK.AAK.BHZ',
      [weavessProps.qcMasksByChannelName[0].id],
      true
    );
    event.shiftKey = true;
    weavessDisplayComponent.onMaskClick(
      event,
      'AAK.AAK.BHZ',
      [weavessProps.qcMasksByChannelName[0].id],
      false
    );
    weavessDisplayComponent.clearSelectedChannels();
    event.shiftKey = false;

    weavessDisplayComponent.onChannelLabelClick(event, 'AAK.AAK.BHZ'); // Select
    expect(weavessDisplayComponent.props.selectedStationIds).toContain('AAK.AAK.BHZ');
    expect(weavessDisplayComponent.props.selectedStationIds).toHaveLength(1);
    weavessDisplayComponent.onChannelLabelClick(event, 'AAK.AAK.BHZ'); // De-select
    expect(weavessDisplayComponent.props.selectedStationIds).toHaveLength(0);
    event.shiftKey = true;
    weavessDisplayComponent.onChannelLabelClick(event, 'AFI.BHZ'); // Range select
    expect(weavessDisplayComponent.props.selectedStationIds).toContain('AFI.BHZ');
    expect(weavessDisplayComponent.props.selectedStationIds).toHaveLength(2);
    event.shiftKey = false;
    event.ctrlKey = true;
    event.altKey = true;
    weavessDisplayComponent.onChannelLabelClick(event, 'AAK.BHZ');
    expect(weavessDisplayComponent.props.selectedStationIds).toContain('AAK.BHZ');
    expect(weavessDisplayComponent.props.selectedStationIds).toContain('AAK.AAK.BHZ');
    expect(weavessDisplayComponent.props.selectedStationIds).toContain('AFI.BHZ');
    expect(weavessDisplayComponent.props.selectedStationIds).toHaveLength(3);

    let boundsGenerator: (
      id: string,
      channelSegment?: WeavessTypes.ChannelSegment
    ) => Promise<WeavessTypes.ChannelSegmentBoundaries>;
    boundsGenerator = weavessDisplayComponent.getBoundariesCalculator(
      AmplitudeScalingOptions.FIXED,
      weavessProps.fixedScaleVal
    );
    let bounds = await boundsGenerator('TEST');
    expect(bounds.channelSegmentId).toBe('TEST');
    expect(bounds.offset).toBe(weavessProps.fixedScaleVal);
    expect(bounds.bottomMax).toBe(-weavessProps.fixedScaleVal);
    expect(bounds.topMax).toBe(weavessProps.fixedScaleVal);
    expect(bounds.channelAvg).toBe(0);
    boundsGenerator = weavessDisplayComponent.getBoundariesCalculator(
      AmplitudeScalingOptions.FIXED,
      'this should trigger scale freezing'
    );
    bounds = await boundsGenerator('TEST');
    expect(bounds.channelSegmentId).toBe('TEST');
    expect(bounds.offset).toBe(weavessProps.fixedScaleVal);
    expect(bounds.bottomMax).toBe(-weavessProps.fixedScaleVal);
    expect(bounds.topMax).toBe(weavessProps.fixedScaleVal);
    expect(bounds.channelAvg).toBe(0);
  });

  test('getWindowedBoundaries gets bounds from the worker cache', async () => {
    const bounds = await weavessDisplayComponent.getWindowedBoundaries('TEST', {
      channelName: 'TEST',
      wfFilterId: 'unfiltered',
      dataSegments: [1, 2, 3, 4]
    });
    expect(bounds).toMatchSnapshot();
    expect(bounds.topMax).toBe(100);
    expect(bounds.bottomMax).toBe(-100);
    expect(bounds.channelSegmentId).toBe('TEST');
  });

  test('getWindowedBoundaries gets bounds from the worker cache when given a time range', async () => {
    const bounds = await weavessDisplayComponent.getWindowedBoundaries(
      'TEST',
      {
        channelName: 'TEST',
        wfFilterId: 'unfiltered',
        dataSegments: [1, 2, 3, 4]
      },
      { startTimeSecs: 0, endTimeSecs: 1000 }
    );
    expect(bounds).toMatchSnapshot();
    expect(bounds.topMax).toBe(100);
    expect(bounds.bottomMax).toBe(-100);
    expect(bounds.channelSegmentId).toBe('TEST');
  });
});
