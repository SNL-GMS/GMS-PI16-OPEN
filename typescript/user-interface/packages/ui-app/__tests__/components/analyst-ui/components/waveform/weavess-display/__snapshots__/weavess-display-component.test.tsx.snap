// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weavess display can mount waveform panel and interact with it 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(WeavessDisplayComponent)
    amplitudeScaleOption="Fixed"
    defaultSignalDetectionPhase="P"
    defaultStations={Array []}
    defaultWaveformFilters={Array []}
    eventsInTimeRange={Array []}
    fixedScaleVal={26}
    measurementMode={
      Object {
        "entries": undefined,
        "mode": "Default",
      }
    }
    qcMasksByChannelName={
      Array [
        Object {
          "channelName": "chanName",
          "currentVersion": Object {
            "category": "REJECTED",
            "channelSegmentIds": Array [],
            "endTime": 1,
            "rationale": "just because",
            "startTime": 0,
            "type": "Spike",
            "version": "version",
          },
          "id": "masky",
          "qcMaskVersions": Array [
            Object {
              "category": "REJECTED",
              "channelSegmentIds": Array [],
              "endTime": 1,
              "rationale": "just because",
              "startTime": 0,
              "type": "Spike",
              "version": "version",
            },
          ],
        },
      ]
    }
    setMeasurementModeEntries={[Function]}
    signalDetectionsByStation={Array []}
    weavessProps={
      Object {
        "customMeasureWindowLabel": undefined,
        "defaultZoomWindow": Object {
          "endTimeSecs": 1274400899,
          "startTimeSecs": 1274391900,
        },
        "endTimeSecs": 1274400899,
        "events": undefined,
        "flex": false,
        "initialConfiguration": Object {
          "defaultChannel": Object {
            "disableMaskModification": true,
            "disableMeasureWindow": true,
            "disableSignalDetectionModification": false,
          },
          "hotKeys": Object {
            "amplitudeScale": "KeyS",
            "amplitudeScaleReset": "Alt+Shift+KeyS",
            "amplitudeScaleSingleReset": "Alt+KeyS",
            "maskCreate": "KeyM",
          },
          "nonDefaultChannel": Object {
            "disableMaskModification": false,
            "disableMeasureWindow": true,
            "disableSignalDetectionModification": true,
          },
          "shouldRenderSpectrograms": false,
          "shouldRenderWaveforms": true,
        },
        "initialZoomWindow": Object {
          "endTimeSecs": 1274399999,
          "startTimeSecs": 1274392800,
        },
        "measureWindowSelection": Object {
          "channel": undefined,
          "endTimeSecs": undefined,
          "isDefaultChannel": undefined,
          "startTimeSecs": undefined,
          "stationId": undefined,
        },
        "selections": Object {
          "channels": Array [],
          "signalDetections": Array [],
        },
        "showMeasureWindow": false,
        "startTimeSecs": 1274391900,
        "stations": Array [],
      }
    }
  >
    <Connect(WeavessDisplayComponent)
      amplitudeScaleOption="Fixed"
      defaultSignalDetectionPhase="P"
      defaultStations={Array []}
      defaultWaveformFilters={Array []}
      eventsInTimeRange={Array []}
      fixedScaleVal={26}
      measurementMode={
        Object {
          "entries": undefined,
          "mode": "Default",
        }
      }
      qcMasksByChannelName={
        Array [
          Object {
            "channelName": "chanName",
            "currentVersion": Object {
              "category": "REJECTED",
              "channelSegmentIds": Array [],
              "endTime": 1,
              "rationale": "just because",
              "startTime": 0,
              "type": "Spike",
              "version": "version",
            },
            "id": "masky",
            "qcMaskVersions": Array [
              Object {
                "category": "REJECTED",
                "channelSegmentIds": Array [],
                "endTime": 1,
                "rationale": "just because",
                "startTime": 0,
                "type": "Spike",
                "version": "version",
              },
            ],
          },
        ]
      }
      reactReduxForwardedRef={null}
      setMeasurementModeEntries={[Function]}
      signalDetectionsByStation={Array []}
      weavessProps={
        Object {
          "customMeasureWindowLabel": undefined,
          "defaultZoomWindow": Object {
            "endTimeSecs": 1274400899,
            "startTimeSecs": 1274391900,
          },
          "endTimeSecs": 1274400899,
          "events": undefined,
          "flex": false,
          "initialConfiguration": Object {
            "defaultChannel": Object {
              "disableMaskModification": true,
              "disableMeasureWindow": true,
              "disableSignalDetectionModification": false,
            },
            "hotKeys": Object {
              "amplitudeScale": "KeyS",
              "amplitudeScaleReset": "Alt+Shift+KeyS",
              "amplitudeScaleSingleReset": "Alt+KeyS",
              "maskCreate": "KeyM",
            },
            "nonDefaultChannel": Object {
              "disableMaskModification": false,
              "disableMeasureWindow": true,
              "disableSignalDetectionModification": true,
            },
            "shouldRenderSpectrograms": false,
            "shouldRenderWaveforms": true,
          },
          "initialZoomWindow": Object {
            "endTimeSecs": 1274399999,
            "startTimeSecs": 1274392800,
          },
          "measureWindowSelection": Object {
            "channel": undefined,
            "endTimeSecs": undefined,
            "isDefaultChannel": undefined,
            "startTimeSecs": undefined,
            "stationId": undefined,
          },
          "selections": Object {
            "channels": Array [],
            "signalDetections": Array [],
          },
          "showMeasureWindow": false,
          "startTimeSecs": 1274391900,
          "stations": Array [],
        }
      }
    >
      <WeavessDisplayComponent
        amplitudeScaleOption="Fixed"
        analysisMode="SCAN"
        currentOpenEventId={null}
        currentTimeInterval={
          Object {
            "endTimeSecs": 1274399999,
            "startTimeSecs": 1274392800,
          }
        }
        defaultSignalDetectionPhase="P"
        defaultStations={Array []}
        defaultWaveformFilters={Array []}
        eventsInTimeRange={Array []}
        fixedScaleVal={26}
        measurementMode={
          Object {
            "entries": undefined,
            "mode": "Default",
          }
        }
        qcMasksByChannelName={
          Array [
            Object {
              "channelName": "chanName",
              "currentVersion": Object {
                "category": "REJECTED",
                "channelSegmentIds": Array [],
                "endTime": 1,
                "rationale": "just because",
                "startTime": 0,
                "type": "Spike",
                "version": "version",
              },
              "id": "masky",
              "qcMaskVersions": Array [
                Object {
                  "category": "REJECTED",
                  "channelSegmentIds": Array [],
                  "endTime": 1,
                  "rationale": "just because",
                  "startTime": 0,
                  "type": "Spike",
                  "version": "version",
                },
              ],
            },
          ]
        }
        sdIdsToShowFk={Array []}
        selectedSdIds={Array []}
        selectedStationIds={Array []}
        setMeasurementModeEntries={[Function]}
        setMode={[Function]}
        setOpenEventId={[Function]}
        setSdIdsToShowFk={[Function]}
        setSelectedSdIds={[Function]}
        setSelectedStationIds={[Function]}
        signalDetectionsByStation={Array []}
        weavessProps={
          Object {
            "customMeasureWindowLabel": undefined,
            "defaultZoomWindow": Object {
              "endTimeSecs": 1274400899,
              "startTimeSecs": 1274391900,
            },
            "endTimeSecs": 1274400899,
            "events": undefined,
            "flex": false,
            "initialConfiguration": Object {
              "defaultChannel": Object {
                "disableMaskModification": true,
                "disableMeasureWindow": true,
                "disableSignalDetectionModification": false,
              },
              "hotKeys": Object {
                "amplitudeScale": "KeyS",
                "amplitudeScaleReset": "Alt+Shift+KeyS",
                "amplitudeScaleSingleReset": "Alt+KeyS",
                "maskCreate": "KeyM",
              },
              "nonDefaultChannel": Object {
                "disableMaskModification": false,
                "disableMeasureWindow": true,
                "disableSignalDetectionModification": true,
              },
              "shouldRenderSpectrograms": false,
              "shouldRenderWaveforms": true,
            },
            "initialZoomWindow": Object {
              "endTimeSecs": 1274399999,
              "startTimeSecs": 1274392800,
            },
            "measureWindowSelection": Object {
              "channel": undefined,
              "endTimeSecs": undefined,
              "isDefaultChannel": undefined,
              "startTimeSecs": undefined,
              "stationId": undefined,
            },
            "selections": Object {
              "channels": Array [],
              "signalDetections": Array [],
            },
            "showMeasureWindow": false,
            "startTimeSecs": 1274391900,
            "stations": Array [],
          }
        }
      >
        <div
          className="weavess-container"
          tabIndex={0}
        >
          <div
            className="weavess-container__wrapper"
          >
            <Blueprint3.NonIdealState
              icon="timeline-line-chart"
              title="No Waveforms to display"
            >
              <div
                className="bp3-non-ideal-state"
              >
                <div
                  className="bp3-non-ideal-state-visual"
                >
                  <Blueprint3.Icon
                    icon="timeline-line-chart"
                    iconSize={60}
                  >
                    <span
                      className="bp3-icon bp3-icon-timeline-line-chart"
                      icon="timeline-line-chart"
                    >
                      <svg
                        data-icon="timeline-line-chart"
                        height={60}
                        viewBox="0 0 20 20"
                        width={60}
                      >
                        <desc>
                          timeline-line-chart
                        </desc>
                        <path
                          d="M19 16H2v-1.59l5-5 3.29 3.29c.18.19.43.3.71.3s.53-.11.71-.29l7-7a1.003 1.003 0 00-1.42-1.42L11 10.59l-3.29-3.3C7.53 7.11 7.28 7 7 7s-.53.11-.71.29L2 11.59V3c0-.55-.45-1-1-1s-1 .45-1 1v14a.998.998 0 001 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
                          fillRule="evenodd"
                          key="0"
                        />
                      </svg>
                    </span>
                  </Blueprint3.Icon>
                </div>
                <Component>
                  <h4
                    className="bp3-heading"
                  >
                    No Waveforms to display
                  </h4>
                </Component>
              </div>
            </Blueprint3.NonIdealState>
          </div>
        </div>
      </WeavessDisplayComponent>
    </Connect(WeavessDisplayComponent)>
  </Connect(WeavessDisplayComponent)>
</Provider>
`;

exports[`weavess display getWindowedBoundaries gets bounds from the worker cache 1`] = `
Object {
  "bottomMax": -100,
  "channelAvg": 0,
  "channelSegmentId": "TEST",
  "offset": 100,
  "samplesCount": 100,
  "topMax": 100,
}
`;

exports[`weavess display getWindowedBoundaries gets bounds from the worker cache when given a time range 1`] = `
Object {
  "bottomMax": -100,
  "channelAvg": 0,
  "channelSegmentId": "TEST",
  "offset": 100,
  "samplesCount": 100,
  "topMax": 100,
}
`;
